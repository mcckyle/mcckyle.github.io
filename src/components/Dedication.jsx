//Filename: Dedication.jsx
//Author: Kyle McColgan
//Date: 27 August 2025
//Description: This file contains the React component for the website dedication section.

import React from 'react';

function Dedication()
{
    return (
        <section id="dedication" aria-label="Dedication">
            <h2>Dedication</h2>
            <p>
                This website is lovingly dedicated to my family and closest friends, whose encouragement fuels my strength.
            </p>
            <p>
                This work is a testament to your guidance, patience, and unwavering faith in me. For all that you are, and all that you have given - I am deeply grateful.
            </p>
        </section>
    );
}

export default Dedication;
