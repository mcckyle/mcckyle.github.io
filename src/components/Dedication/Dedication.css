/* Filename: Dedication.css */

/* =======================
   Dedication Styles.
========================== */
.dedication-title
{
    color: var(--highlight);
}

.dedication-content
{
    font-size: 1.05rem;
    line-height: 1.75;
    color: var(--text);
}

.dedication-content p
{
    margin-bottom: 1rem;
}
