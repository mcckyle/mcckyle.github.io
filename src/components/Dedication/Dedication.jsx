//Filename: Dedication.jsx
//Author: Kyle McColgan
//Date: 2 October 2025
//Description: This file contains the component for the React personal project dedication section.

import React from "react";
import "./Dedication.css";

function Dedication()
{
  return (
    <section
      id="dedication"
      aria-labelledby="dedication-title"
      className="section"
    >
      <h2 id="dedication-title" className="section-title">
        Dedication
      </h2>
      <div id="dedication-body" className="section-body">
        <p>
          This website is lovingly dedicated to my family and closest friends,
          whose encouragement fuels my strength.
        </p>
        <p>
          This work is a testament to your guidance, patience, and unwavering
          faith in me. For all that you are, and all that you have given - I am
          deeply grateful.
        </p>
      </div>
    </section>
  );
}

export default Dedication;
