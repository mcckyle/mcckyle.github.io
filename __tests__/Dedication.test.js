//Filename: Dedication.test.js
//Author: Kyle McColgan
//Date: 29 September 2025
//Description: This file contains the Jest unit tests for my personal website dedication section.

import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Dedication from '../src/components/Dedication/Dedication.jsx';

describe('Dedication component', () => {

  it('renders the Dedication section.', () => {
    render(<Dedication />);
    const section = screen.getByRole('region', { name: /Dedication/i });
    expect(section).toBeInTheDocument();
  });

  it('renders the heading "Dedication".', () => {
    render(<Dedication />);
    const heading = screen.getByRole('heading', { name: /Dedication/i });
    expect(heading).toBeInTheDocument();
  });

  it('renders the section with the correct role.', () => {
    render(<Dedication />);
    const section = screen.getByRole('region');
    expect(section).toHaveAttribute('id', 'dedication');
  });

  it('contains two paragraphs in total.', () => {
    render(<Dedication />);
    const paragraphs = screen.getAllByRole('paragraph');
    expect(paragraphs).toHaveLength(2);
  });

  it('renders the section with a <section> element.', () => {
    render(<Dedication />);
    const section = screen.getByRole('region');
    expect(section.tagName).toBe('SECTION');
  });

  it('does not render any unexpected elements within the section.', () => {
    render(<Dedication />);
    const section = screen.getByRole('region');
    const invalidElements = section.querySelectorAll('script, style');
    expect(invalidElements).toHaveLength(0);
  });

  it('matches the snapshot of the Dedication section.', () => {
    const { asFragment } = render(<Dedication />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the section with the correct ARIA role of "region".', () => {
      render(<Dedication />);
      const section = screen.getByRole('region');
      expect(section).toBeInTheDocument();  // This verifies that the section has the implicit role "region".
  });

  it('does not render empty <p> elements.', () => {
    render(<Dedication />);
    const paragraphs = screen.getAllByRole('paragraph');
    paragraphs.forEach(paragraph => {
      expect(paragraph.textContent).not.toBe('');
    });
  });

  it('renders the section with a valid aria-label.', () => {
    render(<Dedication />);
    const section = screen.getByRole('region');
    expect(section).toHaveAttribute('aria-labelledby', 'dedication-title');
  });

});
