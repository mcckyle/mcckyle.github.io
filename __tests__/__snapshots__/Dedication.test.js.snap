// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dedication component matches the snapshot of the Dedication section. 1`] = `
<DocumentFragment>
  <section
    aria-labelledby="dedication-title"
    class="section"
    id="dedication"
  >
    <div
      class="dedication-content"
    >
      <h2
        class="dedication-title"
        id="dedication-title"
      >
        Dedication
      </h2>
      <p>
        This website is lovingly dedicated to my family and closest friends, whose encouragement fuels my strength.
      </p>
      <p>
        This work is a testament to your guidance, patience, and unwavering faith in me. For all that you are, and all that you have given - I am deeply grateful.
      </p>
    </div>
  </section>
</DocumentFragment>
`;
